### DISCUSSION TITLE
span.ents doesn't identify NER while doc.ents identify it using the same sentence

### DISCUSSION THREAD

**Discussion by sheramir** [explosion/spaCy#7825] [Help: Coding & Implementations]
## How to reproduce the behaviour

I break a text into sentences using 2 methods:
```
sent_list1 = list(s_doc.sents) # list of Span
sent_list2 = [nlp(doc.text) for doc in s_doc.sents]  # list of Doc
```
The lists are identical when comparing the text of the sentences. But the first one (list of Span) misses some named entities.

Full example code:

```
import spacy
from spacy import displacy

story = u'Once upon a time there were three Rabbits, and their names were Flopsy, Mopsy, and Peter.\nPeter was most dreadfully frightened; he rushed all over the garden,for he had forgotten the way back to the gate.'

nlp = spacy.load('en_core_web_sm')
s_doc = nlp(story)
 
sent_list1 = list(s_doc.sents) # list of spacy.tokens.span.Span
sent_list2 = [nlp(doc.text) for doc in s_doc.sents]  # list of spacy.tokens.doc.Doc

type(sent_list1[0])  # output is spacy.tokens.span.Span
type(sent_list2[0]) # output is spacy.tokens.doc.Doc

sent_list2[-1].text == sent_list1[-1].text # True
 
sent_list1[-1].ents # output is []
sent_list2[-1].ents # output is (Peter,)

displacy.render(sent_list1[-1], style="ent",jupyter=True)
# Output is: UserWarning: [W006] No entities to visualize found in Doc object. If this is surprising to you, make sure the Doc was processed using a model that supports named entity recognition, and check the `doc.ents` property manually if necessary.

displacy.render(sent_list2[-1], style="ent",jupyter=True)
#Output is: Peter PERSON was most dreadfully frightened; he rushed all over the garden,
for he had forgotten the way back to the gate.

```

## Your Environment

- **spaCy version:** 3.0.5
- **Platform:** Darwin-20.3.0-x86_64-i386-64bit
- **Python version:** 3.7.9
- **Pipelines:** en_core_web_sm (3.0.0), en_core_web_trf (3.0.0)

**Comment by adrianeboyd** [comment]
Hi, the `ner` component makes predictions for the whole text rather than sentence by sentence. It does consider sentence boundaries so that it doesn't predict an entity span that crosses a sentence boundary, but it takes the context from the preceding/following tokens into account even if they're not in the same sentence.

As a result, processing the sentences one by one can lead to different predictions than processing as a single text because the surrounding context isn't the same for some of the words. For example, sentence-by-sentence the last sentence doesn't contain any context before "Peter", which affects the model predictions.

**Reply by sheramir** [reply]
Thanks for the answer, but I'm still puzzled.
If I look at the `ner` components of the **whole text** `s_doc.ents` I get `(three, Rabbits, Flopsy, Mopsy)`. 'Peter' is not in the ner list at all. That would explain the behaviour you explained. But why isn't 'Peter' identified as `ner` in the whole text, while other (not so common) names are identified? 
If I break the text into separate doc sentences I get 'Peter' as a `ner` in each sentence. But reading the text as a whole wouldn’t identify 'Peter' as `ner`.
It seems like the context of the whole text ruined the `ner` identification, rather than improving it.
